### PR TITLE
Pass transit operator logos to map-popup

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**
Generates and passes a list of transit operator logos to map popup
Offers a shorter more efficient query for the purposes of finding operator logos

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

**Comment by amy-corson-ibigroup** [comment]
Blocked by https://github.com/opentripplanner/otp-ui/pull/775

File: lib/components/util/connected-transit-operator-icons.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import { StopData } from './types'
+import TransitOperatorLogos from './transit-operator-icons'
+
+interface Props {
+  stopData?: StopData
+  transitOperators: TransitOperator[]
+}
+
+class ConnectedTransitOperatorLogos extends Component<Props> {
```
Comment (binh-dam-ibigroup) [inline lib/components/util/connected-transit-operator-icons.tsx]: Convert into a function component (destructure props inside the function declaration).

File: lib/components/util/connected-transit-operator-icons.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  // clean this up
+  return {
+    stopData: stops?.[ownProps.stopId],
+    transitOperators: state.otp.config.transitOperators || []
+  }
+}
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
```
Comment (binh-dam-ibigroup) [inline lib/components/util/connected-transit-operator-icons.tsx]: Delete the unused second parameter, and delete `mapDispatchToProps`.

File: lib/components/util/connected-transit-operator-icons.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+const mapDispatchToProps = {}
+
+const mapStateToProps = (
+  state: AppReduxState,
+  ownProps: Props & { stopId: string }
+) => {
+  const stops = state.otp.transitIndex.stops
+  // clean this up
```
Comment (binh-dam-ibigroup) [inline lib/components/util/connected-transit-operator-icons.tsx]: Remove this comment.

File: lib/components/util/transit-operator-icons.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      // e.g. "sound-transit"
+      <span
+        className={
+          operator.name ? operator.name.replace(/\s+/g, '-').toLowerCase() : ''
+        }
+      >
+        <OperatorLogo
+          alt={operatorLogoAriaLabel}
+          marginRight={0.5}
+          maxHeight={1}
```
Comment (binh-dam-ibigroup) [inline lib/components/util/transit-operator-icons.tsx]: If this is the only place `marginRight` and `maxHeight` are passed, could it be better to do a `styled` `OperatorLogo` or `OperatorImg` instead?

File: lib/components/map/default-map.tsx [hunk 5]
```diff
@@ -34,6 +36,7 @@ import RoutePreviewOverlay from './route-preview-overlay'
 import RouteViewerOverlay from './connected-route-viewer-overlay'
 import StopsOverlay from './connected-stops-overlay'
 import TransitiveOverlay from './connected-transitive-overlay'
+import TransitOperatorLogos from '../util/connected-transit-operator-icons'
```
Comment (binh-dam-ibigroup) [inline lib/components/map/default-map.tsx]: Rename to `TransitOperatorIcons`.

File: lib/components/util/connected-transit-operator-icons.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/util/connected-transit-operator-icons.tsx]: Rename to `TransitOperatorIcons`.

File: lib/components/util/transit-operator-icons.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  }
+}
+
+const TransitOperatorLogos = ({
+  loading = false,
+  stopData,
+  transitOperators
+}: {
+  loading?: boolean
+  stopData: StopData | undefined
```
Comment (binh-dam-ibigroup) [inline lib/components/util/transit-operator-icons.tsx]: Use `?:` to denote that a prop can also be `undefined`.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Looks good, but please tidy up as indicated.

File: lib/components/util/transit-operator-icons.tsx (hunk 4 above)
Comment (amy-corson-ibigroup) [reply lib/components/util/transit-operator-icons.tsx]: Good idea!

File: lib/components/util/transit-operator-icons.tsx (hunk 6 above)
Comment (amy-corson-ibigroup) [reply lib/components/util/transit-operator-icons.tsx]: 🤦‍♀️

File: lib/components/util/operator-logo.tsx [hunk 7]
```diff
@@ -6,35 +6,28 @@ const OperatorImg = styled.img<{ marginRight?: number; maxHeight?: number }>`
...
-  return (
-    <OperatorImg
-      alt={alt || operator.name}
-      marginRight={marginRight}
-      maxHeight={maxHeight}
-      src={operator.logo}
-    />
-  )
+  if (styled)
+    return <StyledOperatorImg alt={alt || operator.name} src={operator.logo} />
```
Comment (binh-dam-ibigroup) [inline lib/components/util/operator-logo.tsx:28]: Surround with curly braces.

**Review by binh-dam-ibigroup — APPROVED** [review]
Cool, thanks for the changes, just one style tweak.

**Review by josh-willis-arcadis — APPROVED** [review]
Great work!

**Pull request merged**